aptos-move/aptos-vm/src/block_executor/mod.rs

```diff
@@ -39,47 +41,80 @@ impl BlockExecutorTransaction for PreprocessedTransaction {
 }
 
 // Wrapper to avoid orphan rule
-#[derive(PartialEq, Debug)]
-pub(crate) struct AptosTransactionOutput(TransactionOutputExt);
+#[derive(Debug)]
+pub(crate) struct AptosTransactionOutput {
+    output_ext: Mutex<Option<TransactionOutputExt>>,
+    output_with_deltas: OnceCell<TransactionOutput>,
+}
 
 impl AptosTransactionOutput {
-    pub fn new(output: TransactionOutputExt) -> Self {
-        Self(output)
+    pub(crate) fn new(output: TransactionOutputExt) -> Self {
+        Self {
+            output_ext: Mutex::new(Some(output)),
+            output_with_deltas: OnceCell::new(),
```

Comment:
Can we maybe call it `materialized_output` or `committed_output`? To me `output_with_deltas` sounds like there is and output and it has deltas separately